File: back-end/src/main/java/es/ull/project/domain/valueobject/NodeName.java
package es.ull.project.domain.valueobject;

import java.util.Objects;

/**
 * @brief Value object representing the name of a Node.
 */
public final class NodeName {

    private static final String ERROR_NAME_NOT_DEFINED = "Node name not defined";
    private final String name;

    public NodeName(String name) {
        this.validateName(name);
        this.name = name.trim();
    }

    private void validateName(String name) {
        if (name == null || name.trim().isEmpty()) {
            throw new IllegalArgumentException(ERROR_NAME_NOT_DEFINED);
        }
    }

    public String getName() {
        return this.name;
    }

    @Override
    public boolean equals(Object otherObject) {
        if (this == otherObject) {
            return true;
        }
        if (!(otherObject instanceof NodeName)) {
            return false;
        }
        final NodeName other = (NodeName) otherObject;
        return this.name.equalsIgnoreCase(other.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name.toLowerCase());
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: front-end/src/domain/value-object/node-name.ts
export class NodeName {
    private static readonly ERROR_NAME_NOT_DEFINED = "Node name not defined";
    private readonly name: string;

    constructor(name: string) {
        this.validateName(name);
        this.name = name;
    }

    getName(): string {
        return this.name;
    }

    setName(name: string): NodeName {
        this.validateName(name);
        return new NodeName(name);
    }


    validateName(name: string): void {
        if (!name || name.trim().length === 0) {
            throw new Error(NodeName.ERROR_NAME_NOT_DEFINED);
        }
    }

    equals(other: NodeName): boolean {
        return other instanceof NodeName && this.name === other.name;
    }

    toString(): string {
        return `NodeName={${this.name}}`;
    }
}
